src/Corax/Utils/GrowableBuffer.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using Sparrow.Server;
+namespace Corax.Utils;
+
+public interface IBufferGrowth
+{
+    public int GetInitialSize(in long initialSize);
+    public int GetNewSize(in int currentSizeInBytes);
+
+    public bool GrowingThresholdExceed(in int count, in int sizeInBytes);
+}
+
+public readonly struct Slowly : IBufferGrowth
+{
+    public int GetNewSize(in int currentSizeInBytes)
+    {
+        var size = currentSizeInBytes > 16 * Sparrow.Global.Constants.Size.Megabyte
+            ? (int)(currentSizeInBytes * 1.5)
+            : currentSizeInBytes * 2;
+
+        // Represent array as N*sizeof(long)
+        return size - (size % sizeof(long));
+    }
+
+    public bool GrowingThresholdExceed(in int count, in int sizeInBytes)
+    {
+        // 1/16 left
+        var amountOfLongs = (sizeInBytes / sizeof(long));
+        return (amountOfLongs - count) < amountOfLongs / 16;
+    }
+
+    public int GetInitialSize(in long initialSize)
+    {
+        var size = 4 * Math.Min(Math.Max(Sparrow.Global.Constants.Size.Kilobyte, (int)initialSize), 16 * Sparrow.Global.Constants.Size.Kilobyte);
+        // Represent array as N*sizeof(long)
+        return size - (size % sizeof(long));
+    }
+}
+
```

Comment:
Maybe we should also add a conservative policy. Starts with just a few elements (16) and grow from there.